main.tf

```diff
@@ -34,49 +38,65 @@ module "spacelift_environment" {
   terraform_version = var.terraform_version
   autodeploy        = var.autodeploy
 
-  terraform_version_map = var.terraform_version_map
+  terraform_version_map        = var.terraform_version_map
+  process_component_stack_deps = var.process_component_stack_deps
 }
 
 # Define the global trigger policy that allows us to trigger on various context-level updates
 resource "spacelift_policy" "trigger_global" {
-  type = "TRIGGER"
+  count = var.trigger_global_enabled ? 1 : 0
 
+  type = "TRIGGER"
   name = "Global Trigger Policy"
   body = file("${path.module}/policies/trigger-global.rego")
 }
```

Comment:
General thought: Maybe instead of toggling whether or not we _create_ these policies, we toggle whether or not we _attach_ these policies to the stacks by default. This way people could manually attach the policies on occasion.